examples/04_react_router/src/App.tsx

```diff
@@ -0,0 +1,55 @@
+import {
+  unstable_HistoryRouter as BrowserRouter,
+  Routes,
+  Route,
+  Link,
+} from 'react-router-dom';
+import { atomWithLocation } from 'jotai-location';
+import { useAtomValue } from 'jotai';
+import * as React from 'react';
+import history from './routerHistory';
+
+const Route1 = <h1>Hello</h1>;
+const Route2 = <h1>World</h1>;
+const location = atomWithLocation();
+location.onMount = (set) => {
```

Comment:
I was hoping to see using `subscribe` and other options in atomWithLocation, and avoiding `onMount`.
Can you try that? Let me know if you need any help.

Reply:
How about something like this?
```js
const location = atomWithLocation({
  loc: { searchParams: new URLSearchParams(history.location.search), ...history.location },
  getLocation() {
    return this loc;
  },
  subscribe(arg) {
    const callback = (callbackArg) => {
      const searchParams = new URLSearchParams(callbackArg.location.search);
      const loc = { searchParams, ...callbackArg.location };
      this.loc = loc;
    };
    const unlisten = history.listen((listenArg) => {});
    callback(history);
    return unlisten;
  },
});
```

If `callbackArg` is equal to `history`, it would be easier.

Reply:
I don't see how you would get this to work with the current implementation of atomWithLocation. Do you think this can be done without changing the implementation?

Reply:
It's not difficult to avoid `this`, but can you try this? Doesn't it work?
```js
const location = atomWithLocation({
  getLocation: () => ({
    searchParams: new URLSearchParams(history.location.search),
    ...history.location,
  }),
  subscribe: (callback) => history.listen(callback),
});
```

Reply:
Seems to be working great, and is very readable and concise as well! 🤩

Reply:
I wonder if you need to pass applyLocation. Or does it work without it?

Reply:
Seems to be working fine without it! 👌 
https://codesandbox.io/p/sandbox/jotai-location-not-updating-forked-x9gmft?file=%2Fsrc%2FApp.js%3A15%2C36